ILSpy.ReadyToRun/ReadyToRunLanguage.cs

```diff
@@ -187,32 +277,98 @@ private void Disassemble(PEFile currentFile, ITextOutput output, ReadyToRunReade
-				output.Write($" ; {unwindCode.UnwindOp}({unwindCode.OpInfoStr})");
+				foreach (var unwindCode in unwindInfo[nextInstructionOffset]) {
+					output.Write($" ; {unwindCode.UnwindOp}({unwindCode.OpInfoStr})");
+				}
 			}
 		}
 
+		private static void DecorateDebugInfo(ITextOutput output, Instruction instr, Dictionary<VarLocType, HashSet<Tuple<DebugInfo, NativeVarInfo>>> debugInfoDict, ulong baseInstrIP)
+		{
+			if (debugInfoDict != null) {
+				InstructionInfoFactory factory = new InstructionInfoFactory();
+				InstructionInfo info = factory.GetInfo(instr);
+				HashSet<Tuple<DebugInfo, NativeVarInfo>> stkSet = new HashSet<Tuple<DebugInfo, NativeVarInfo>>();
+				if (debugInfoDict.ContainsKey(VarLocType.VLT_STK)) {
+					stkSet.UnionWith(debugInfoDict[VarLocType.VLT_STK]);
+				}
+				if (debugInfoDict.ContainsKey(VarLocType.VLT_STK_BYREF)) {
+					stkSet.UnionWith(debugInfoDict[VarLocType.VLT_STK_BYREF]);
+				}
+				if (stkSet != null) {
+					foreach (UsedMemory usedMemInfo in info.GetUsedMemory()) { //for each time a [register +- value] is used
+						foreach (Tuple<DebugInfo, NativeVarInfo> tuple in stkSet) { //for each VLT_STK variable
+							var debugInfo = tuple.Item1;
+							var varInfo = tuple.Item2;
+							int stackOffset = varInfo.VariableLocation.Data2;
+							ulong adjOffset;
+							bool negativeOffset;
+							if (stackOffset < 0) {
+								int absValue = -1 * stackOffset;
+								adjOffset = ulong.MaxValue - (ulong)absValue + 1;
+								negativeOffset = true;
+							} else {
+								adjOffset = (ulong)stackOffset;
+								negativeOffset = false;
+							}
+							if (varInfo.StartOffset < instr.IP - baseInstrIP && varInfo.EndOffset > instr.IP - baseInstrIP &&
+								DebugInfo.GetPlatformSpecificRegister(debugInfo.Machine, varInfo.VariableLocation.Data1) == usedMemInfo.Base.ToString() &&
+								adjOffset == usedMemInfo.Displacement) {
+
+								output.Write($"; [{usedMemInfo.Base.ToString()}{(negativeOffset ? '-' : '+')}{Math.Abs(stackOffset)}] = {varInfo.Variable.Type} {varInfo.Variable.Index}");
+							}
+						}
+					}
+				}
+				HashSet<Tuple<DebugInfo, NativeVarInfo>> regSet = new HashSet<Tuple<DebugInfo, NativeVarInfo>>();
+				if (debugInfoDict.ContainsKey(VarLocType.VLT_REG)) {
+					regSet.UnionWith(debugInfoDict[VarLocType.VLT_REG]);
+				}
+				if (debugInfoDict.ContainsKey(VarLocType.VLT_REG_BYREF)) {
+					regSet.UnionWith(debugInfoDict[VarLocType.VLT_REG_BYREF]);
+				}
+				if (debugInfoDict.ContainsKey(VarLocType.VLT_REG_FP)) {
+					regSet.UnionWith(debugInfoDict[VarLocType.VLT_REG_FP]);
+				}
+				if (regSet != null) {
+					foreach (UsedRegister usedMemInfo in info.GetUsedRegisters()) {
+						foreach (Tuple<DebugInfo, NativeVarInfo> tuple in regSet) {
+							var debugInfo = tuple.Item1;
+							var varInfo = tuple.Item2;
+							if (varInfo.StartOffset < instr.IP - baseInstrIP && varInfo.EndOffset > instr.IP - baseInstrIP &&
```

Comment:
I need to double-check with this, I think the `StartOffset` should be inclusive, the `EndOffset` I am not sure.